File: practice7/src/Owl.java
public class Owl extends Animals{
    private boolean isDomesticated;
    private int hunger;

    public Owl(boolean isDomesticated, int hunger) {
        this.isDomesticated = isDomesticated;
        this.hunger = hunger;
    }

    @Override
    public void voice() {

    }

    @Override
    public void feed(int hunger) {

    }

    public boolean isDomesticated() {
        return isDomesticated;
    }

    public void setDomesticated(boolean domesticated) {
        isDomesticated = domesticated;
    }

    public int getHunger() {
        return hunger;
    }

    public void setHunger(int hunger) {
        this.hunger = hunger;
    }

    @Override
    public String toString() {
        return "Owl{" + name + age +
                "isDomesticated=" + isDomesticated +
                ", hunger=" + hunger +
                '}';
    }
}

File: PythonProject/Owl.py
from Animal import Animal
from Salmon import Salmon


class Owl(Animal):
    hunger = 0

    def __init__(self, name, age) -> None:
        super().__init__(name, age)
    def voice(self):
        print("u-u, u-u")

    def feed(self, hunger, food):
        if (isinstance(food, Salmon)): self.hunger -= hunger
        else: print("boo")

    def addHunger(self, hunger):
        self.hunger += hunger
    def isHungry(self):
        if (self.hunger > 0):
            print("I'am hungry")
        else:
            print("I'am NOT hungry")
